src/devices/BuildHat/Sensors/Sensor.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using Iot.Device.BuildHat.Models;
+
+namespace Iot.Device.BuildHat.Sensors
+{
+    /// <summary>
+    /// Interface for a sensor
+    /// </summary>
+    public class Sensor
+    {
+        internal Brick Brick;
+
+        /// <summary>
+        /// Gets true if the motor is connected.
+        /// </summary>
+        public bool IsConnected { get; internal set; }
+
+        /// <summary>
+        /// Creates a sensor
+        /// </summary>
+        /// <param name="brick">The brick.</param>
+        /// <param name="port">The port.</param>
+        /// <param name="type">The sensor type.</param>
+        internal Sensor(Brick brick, SensorPort port, SensorType type)
```

Comment:
what would that bring if it's protected compare to the current implementation?
All the other sensors, including the motors are heritating from it. All the others should be protected as well? I'm happy to adjust if needed of course!

Reply:
If you make it protected, it would be possible to externally add additional sensors. Like this, it is not possible to externally inherit from this class. 

Also, if you don't need instances of this class itself, it should probably be abstract.

Reply:
ok, good idea then to have it protected. Someone can build its own version from this.

Reply:
Good. Then please review the other members for external usage as well, please (e.g. you probably want the events to be public or protected, too)